### ISSUE TITLE
OpenAPI v2 converter - Missing Parameter.style property

### ISSUE THREAD

**Issue by ackintosh** [swagger-api/swagger-parser#690]
#### sample spec (OpenAPI v2) includes the parameters which has 'collectionFormat'

```json
{
    "swagger": "2.0",
    "info": {
        "version": "1.0",
        "title": "x-example"
    },
    "host": "httpbin.org",
    "basePath": "/anything",
    "schemes": [
        "http"
    ],
    "paths": {
        "/{foo}": {
            "get": {
                "parameters": [
                  {
                    "name": "id",
                    "in": "path",
                    "description": "ID of pet to use",
                    "required": true,
                    "type": "array",
                    "items": {
                      "type": "string"
                    },
                    "collectionFormat": "csv"         <- CSV
                  },
                  {
                    "name": "enum_header_string_array",
                    "in": "header",
                    "description": "",
                    "type": "array",
                    "items": {
                      "type": "string",
                      "enum": [
                        ">",
                        "$"
                      ]
                    },
                    "collectionFormat": "csv"         <- CSV
                  }
                ],
                "responses": {
                  "200": {
                    "description": "OK"
                  }
                }
            }
        }
    }
}
```

#### the parameters are converted as 'style: null'

```java
    public void testIssueXXX() throws Exception {
        OpenAPI oas = getConvertedOpenAPIFromJsonFile("issue-xxx.json");
        List<Parameter> parameters = oas.getPaths().get(FOO_PATH).getGet().getParameters();

        Parameter param = parameters.get(0);
        System.out.println(param.toString());

        Parameter param2 = parameters.get(1);
        System.out.println(param2.toString());

    }

// class Parameter {
//     name: id
//     in: path
//     description: ID of pet to use
//     required: true
//     deprecated: null
//     allowEmptyValue: null
//     style: null                  <<-- "style" is null



// class Parameter {
//     name: enum_header_string_array
//     in: header
//     description: null
//     required: null
//     deprecated: null
//     allowEmptyValue: null
//     style: null                  <<-- "style" is null
```

---

I think the 'style' properties should be 'simple' in case of the example. 💡 I'll submit a PR to fix the issue.

**Comment by jmini** [comment]
My understanding of [3.0.1 Spec: Style values section](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.1.md#style-values).


* For query & cookie parameters:
  - `collectionFormat` == `csv` is converted to `style`: `form` & `explode`: `false`
  - `collectionFormat` == `multi` is converted to `style`: `form` & `explode`: `true`
* For path & header parameters
  -  `collectionFormat` == `csv` is converted to `style`: `simple`

**Comment by wing328** [comment]
I think this is the same as https://github.com/wing328/openapi-generator/issues/72 so we will need to wait for the fix in Swagger Parser.

A workaround is to convert the spec to OAS3 which seems to work well with "style", "form" at the moment.

**Comment by ackintosh** [comment]
@jmini Thanks! ✨ 
Here is the summary I think. Please let me know if you recognize anything wrong. 🙏 

|OAS v2: in  |OAS v2: collectionFormat  |OAS v3: style  |OAS v3: explode|
|---|---|---|---|
|query  |csv  |form  |false|
|query  |multi  |form  |true|
|header  |csv  |simple  |false(default)|
|path  |csv  |simple  |false(default)|
|formData  |*  |form  |true|
|*  |ssv  |spaceDelimited  |false(default)|
|*  |pipes  |pipeDelimited  |false(default)|
|*  |tsv  |-  |-|

Note: In OAS v2, `collectionFormat: multi` is valid only for "query" or "formData".
https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#parameter-object

**Comment by frantuma** [comment]
fixed in #711, please reopen if still experiencing issues
